lightkurve/correctors/pldcorrector.py

```diff
@@ -82,199 +116,509 @@ class PLDCorrector(object):
+            model.gp = xo.gp.GP(kernel, lc_time, diag + tt.exp(logs2))
+            model.cadence_mask = cadence_mask
+
+            # The motion model regresses against the design matrix
+            A = tt.dot(design_matrix.T, model.gp.apply_inverse(design_matrix))
+            B = tt.dot(design_matrix.T, model.gp.apply_inverse(lc_flux[:, None]))
+            weights = tt.slinalg.solve(A, B)
+            motion_model = pm.Deterministic("motion_model", tt.dot(design_matrix, weights)[:, 0])
+            pm.Deterministic("weights", weights)
+
+            # Likelihood to optimize
+            pm.Potential("obs", model.gp.log_likelihood(lc_flux - motion_model))
+
+            # Track the flux values of the different diagnostic light curves for ease of use
+            gp_model, gp_model_var = model.gp.predict(return_var=True)
+            pm.Deterministic("gp_model", gp_model)
+            pm.Deterministic("gp_model_std", np.sqrt(gp_model_var))
+            pm.Deterministic("corrected_flux",
+                             lc_flux - motion_model + tt.mean(motion_model))
+            pm.Deterministic("corrected_flux_without_gp",
+                             lc_flux - motion_model - gp_model + tt.mean(motion_model))
+
+        self.most_recent_model = model
+        return model
+
+    def optimize(self, model=None, start=None, robust=False, **kwargs):
+        """Returns the maximum likelihood solution.
+
+        Parameters
+        ----------
+        model : `pymc3.model.Model` object
+            A pymc3 model.  If `None`, the the output of this object's
+            `create_pymc_model` method will be used.
+        start : dict
+            MAP Solution from exoplanet
+        robust : bool
+            If `True`, all parameters will be optimized separately before
+            attempting to optimize all parameters together.  This will be
+            significantly slower but increases the likelihood of success.
+        **kwargs : dict
+            Dictionary of arguments to be passed to
+            `~lightkurve.correctors.PLDCorrector.create_pymc_model`.
 
+        Returns
+        -------
+        solution : dict
+            Maximum likelihood values.
+        """
+        if model is None:
+            model = self.create_pymc_model(**kwargs)
+        if start is None:
+            start = model.test_point
+
+        with model:
+            solution = xo.optimize(start=start, vars=[model.logs2])
+            # Optimizing parameters separately appears to make finding a solution more likely
+            if robust:
+                solution = xo.optimize(start=solution, vars=[model.logsigma])
+                solution = xo.optimize(start=solution, vars=[model.logrho, model.logsigma])
+                solution = xo.optimize(start=solution, vars=[model.logs2])
+            solution = xo.optimize(start=solution)  # Optimize all parameters
+
+        self.most_recent_solution_or_trace = solution
```

Comment:
Should there be `most_recent_solution` and `most_recent_trace`? The two are quite different.